service/share/share.go

```diff
@@ -145,16 +146,52 @@ func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID name
 	}
 
 	namespacedShares := make([]Share, 0)
+	nodeCh := make(chan []format.Node)
+	errCh := make(chan error)
```

Comment:
The code can be simplified by making an anonymous `result struct` that has both the value and the error, so only one channel of the type needs to be created and handled. That's a common pattern in such situations.

Reply:
done